pending_txs/projector.go

```diff
@@ -0,0 +1,49 @@
+package pending_txs
```

Comment:
It's probably too late to raise this issue so I'm just gonna put this here as a FYI and for future discussion:
As we've discussed with Lane, I think this solution is redundant and quite complicated for the given task. 
As an alternative and more simple solution is to run the transactions to be included in a block in a replica of the current state and use same logic that processes transactions from a block. basically replicating an instance of the global state and the processor. This has some advantages since there is no replication or different logic in processing txs and accepting txs into a block. This will come in handy when introducing smart contracts since we can't have different logic in accepting and running tx/code.

Reply:
Your suggested solution assumes that when we receive a new block (or when creating a new block), we know for sure what will be the state when the block is applied. This is not the case in a mesh, therefore we need a way to maintain several fututre "states" with multiple txs per nonce.

Reply:
I don't understand what do you mean by several future states? There is one agreed state and all transasctions can be tested on that state. If you wish to insert multiple transactions from same origin, you can do so, and still test inserting the ordered by nonce to the same state... same way they would be applied by transaction processor when they are read from a block. the main issue is not to have 2 algorithm determining whether a trasaction is accepted or not.